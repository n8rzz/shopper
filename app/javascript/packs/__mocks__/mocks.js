import { assemblyItemsGroupedByAssemblyIdMock } from './assembly-item.mocks';
import { assemblyMock, assemblyListMock, assemblyMapMock } from './assembly.mock';
import { csrfMock } from './csrf.mock';
import { departmentListMock, departmentMapMock } from './department.mock';
import { itemMock, itemListMock, itemMapMock } from './item.mock';
import { locationDepartmentListMock } from './location-department.mock';
import { locationMock } from './location.mock';
import { orderItemListMock } from './order-item.mock';
import { orderMock, orderLocationNameMock } from './order.mock';

export {
    assemblyItemsGroupedByAssemblyIdMock,
    assemblyMock,
    assemblyListMock,
    assemblyMapMock,
    csrfMock,
    departmentListMock,
    departmentMapMock,
    itemMock,
    itemListMock,
    itemMapMock,
    locationDepartmentListMock,
    locationMock,
    orderItemListMock,
    orderMock,
    orderLocationNameMock,
};
