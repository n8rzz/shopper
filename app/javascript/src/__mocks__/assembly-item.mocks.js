export const assemblyItemsGroupedByAssemblyIdMock = {
    1: [
        {
            assembly_id: 1,
            created_at: '2019-08-01T22:58:09.157Z',
            id: 1,
            item_id: 13,
            qty: 8,
            updated_at: '2019-08-01T22:58:09.157Z',
        }
    ],
};
