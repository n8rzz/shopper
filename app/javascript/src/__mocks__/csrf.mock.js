export const csrfMock = '3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ==';
