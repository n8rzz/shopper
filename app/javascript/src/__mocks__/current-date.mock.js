export const currentDateMock = new Date('09/01/2019');
