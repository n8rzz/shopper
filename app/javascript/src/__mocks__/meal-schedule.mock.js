import _groupBy from 'lodash/groupBy';

export const mealScheduleItemMock = {
    id: 1,
    schedule_date: '2019-09-24T00:00:00.000Z',
    purchased: false,
    created_at: '2019-09-04T01:00:35.645Z',
    updated_at: '2019-09-04T01:00:35.645Z',
    item_id: 15,
    order_item_id: null,
    assembly_id: null,
    meal_time: 'none',
};

export const mealScheduleAssemblyMock = {
    id: 2,
    schedule_date: '2019-09-24T00:00:00.000Z',
    purchased: false,
    created_at: '2019-09-04T01:00:35.645Z',
    updated_at: '2019-09-04T01:00:35.645Z',
    item_id: null,
    order_item_id: null,
    assembly_id: 2,
    meal_time: 'none',
};

export const mealScheduleListMock = [
    mealScheduleItemMock,
    mealScheduleAssemblyMock,
    {
        id: 3,
        schedule_date: '2019-09-29T00:00:00.000Z',
        purchased: false,
        created_at: '2019-09-03T01:00:35.645Z',
        updated_at: '2019-09-07T01:00:35.645Z',
        item_id: null,
        order_item_id: null,
        assembly_id: 2,
        meal_time: 'none',
    },
];

export const mealScheduleMapMock = _groupBy(mealScheduleListMock, 'schedule_date');
