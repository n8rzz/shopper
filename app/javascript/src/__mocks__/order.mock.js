export const orderMock = {
    created_at: '2019-08-06T22:19:49.806Z',
    id: 2,
    location_id: 8,
    shopping_date: '2019-11-05T00:00:00.000Z',
    status: 'active',
    updated_at: '2019-08-06T22:19:49.806Z',
};

export const orderLocationNameMock = 'Durgan, Mills and Labadie';
