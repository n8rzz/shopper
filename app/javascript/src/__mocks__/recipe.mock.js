export const recipeMock = {
    id: 1,
    name: 'Mushroom Risotto',
    description: 'Two buttermilk waffles, topped with whipped cream and maple syrup, a side of two eggs served any style, and your choice of smoked bacon or smoked ham.',
    prep_time: 14,
    cook_time: 18,
    yield_value: 28,
    yield_unit: 'servings',
    ownable_type: 'Group',
    ownable_id: 1,
    created_at: '2020-03-01T04:13:06.254Z',
    updated_at: '2020-03-01T04:13:06.254Z',
    assembly_id: null,
};

export const recipeWithAssemblyMock = {
    id: 2,
    name: 'Mushroom Risotto',
    description: 'Two buttermilk waffles, topped with whipped cream and maple syrup, a side of two eggs served any style, and your choice of smoked bacon or smoked ham.',
    prep_time: 14,
    cook_time: 18,
    yield_value: 28,
    yield_unit: 'servings',
    ownable_type: 'Group',
    ownable_id: 1,
    created_at: '2020-03-01T04:13:06.254Z',
    updated_at: '2020-03-01T04:13:06.254Z',
    assembly_id: 1,
};

export const recipeListMock = [
    recipeMock,
    recipeWithAssemblyMock,
];
