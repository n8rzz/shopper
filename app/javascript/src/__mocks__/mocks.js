import { assemblyItemsGroupedByAssemblyIdMock } from './assembly-item.mocks';
import { assemblyMock, assemblyListMock, assemblyMapMock } from './assembly.mock';
import { csrfMock } from './csrf.mock';
import { currentDateMock } from './current-date.mock';
import { departmentListMock, departmentMapMock } from './department.mock';
import { itemMock, itemListMock, itemMapMock } from './item.mock';
import { locationDepartmentListMock } from './location-department.mock';
import { locationMock } from './location.mock';
import {
    mealScheduleAssemblyMock,
    mealScheduleItemMock,
    mealScheduleListMock,
    mealScheduleMapMock,
} from './meal-schedule.mock';
import { orderItemListMock } from './order-item.mock';
import { orderMock, orderLocationNameMock } from './order.mock';

export {
    assemblyItemsGroupedByAssemblyIdMock,
    assemblyMock,
    assemblyListMock,
    assemblyMapMock,
    csrfMock,
    currentDateMock,
    departmentListMock,
    departmentMapMock,
    itemMock,
    itemListMock,
    itemMapMock,
    locationDepartmentListMock,
    locationMock,
    mealScheduleAssemblyMock,
    mealScheduleItemMock,
    mealScheduleListMock,
    mealScheduleMapMock,
    orderItemListMock,
    orderMock,
    orderLocationNameMock,
};
