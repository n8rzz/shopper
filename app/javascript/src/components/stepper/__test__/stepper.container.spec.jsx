import React from 'react';
import sinon from 'sinon';
import { mount, shallow } from 'enzyme';

import { StepperContainer } from '../stepper.container';
import { csrfMock, itemMock } from '../../../__mocks__/mocks';

describe('StepperContainer', () => {
    let component = null;
    let onSubmitHandlerSpy = null;

    beforeEach(() => {
        onSubmitHandlerSpy = sinon.spy();
        component = mount(<StepperContainer
            csrf={csrfMock}
            departmentId={itemMock.department_id}
            itemId={itemMock.id}
            itemName={itemMock.name}
            isSubmitting={false}
            isSubmitSuccess={false}
            onSubmitHandler={onSubmitHandlerSpy}
        />);
    });

    afterEach(() => {
        onSubmitHandlerSpy = null;
        component = null;
    });

    test('renders correctly with valid props', () => {
        const shallowComponent = shallow(<StepperContainer
            csrf={csrfMock}
            departmentId={itemMock.department_id}
            itemId={itemMock.id}
            itemName={itemMock.name}
            isSubmitting={false}
            isSubmitSuccess={false}
            onSubmitHandler={sinon.spy()}
        />);

        expect(shallowComponent).toMatchSnapshot();
    });

    describe('decrease btn element', () => {
        test('contains decrease btn element', () => {
            const decreaseBtn = component.find('.js-stepper-btn-decrease');

            expect(decreaseBtn.length).toBe(1);
        });

        test('responds to click', () => {
            const instance = component.instance();
            component.setState({ qty: 2 });

            instance._onClickDecrease();

            expect(component.state('qty')).toBe(1);
        });

        test('does not decrease passed 1', () => {
            const instance = component.instance();
            component.setState({ qty: 1 });

            instance._onClickDecrease();

            expect(component.state('qty')).toBe(1);
        });
    });

    describe('increase btn element', () => {
        test('contains increase btn element', () => {
            const increaseBtn = component.find('.js-stepper-btn-increase');

            expect(increaseBtn.length).toBe(1);
        });

        test('responds to click', () => {
            const instance = component.instance();

            instance._onClickIncrease();

            expect(component.state('qty')).toBe(2);
        });
    });

    describe('submit element', () => {
        test('contains submit btn element', () => {
            const submitBtn = component.find('.stepper-ft-submit');

            expect(submitBtn.length).toBe(1);
        });

        test('responds to click', () => {
            const eventMock = {
                stopPropagation: sinon.spy(),
                preventDefault: sinon.spy(),
            };
            const instance = component.instance();

            instance._onClickSubmit(eventMock);

            expect(onSubmitHandlerSpy.calledWith(component.state('qty'))).toBe(true);
        });

        test('returns early if #qty is < 1', () => {
            const eventMock = {
                stopPropagation: sinon.spy(),
                preventDefault: sinon.spy(),
            };
            const instance = component.instance();

            component.setState({ qty: 0 });
            instance._onClickSubmit(eventMock);

            expect(onSubmitHandlerSpy.callCount).toBe(0);
        });

        test('resets #qty to 1 after submit', () => {
            const eventMock = {
                stopPropagation: sinon.spy(),
                preventDefault: sinon.spy(),
            };
            const instance = component.instance();

            component.setState({ qty: 3 });
            instance._onClickSubmit(eventMock);

            expect(component.state('qty')).toBe(1);
        });
    });
});
