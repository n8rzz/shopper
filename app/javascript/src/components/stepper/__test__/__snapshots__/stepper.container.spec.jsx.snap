// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepperContainer renders correctly with valid props 1`] = `
<Stepper
  isSubmitSuccess={false}
  isSubmitting={false}
  itemId={15}
  onClickDecreaseHandler={[Function]}
  onClickIncreaseHandler={[Function]}
  onClickSubmitHandler={[Function]}
  qty={1}
/>
`;
